client/src/lib.rs

```diff
@@ -684,9 +686,11 @@ pub trait WorkflowClientTrait {
         task_queue: String,
         workflow_id: String,
         workflow_type: String,
+        request_id: Option<String>,
```

Comment:
This argument list is starting to get pretty outta control.

I'm thinking it makes sense to accept arg structs here and make builders for them or something like that. Not in this PR I think, but, looks like that's starting to become a good idea.

Reply:
agreed, shouldn't be too bad with derive_builder